generators/server/templates/gradle/_docker.gradle

```diff
@@ -16,24 +16,31 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 -%>
-import org.gradle.internal.os.OperatingSystem
-
-task buildDocker(type: Exec) {
-    doFirst {
-        copy {
-            from "src/main/docker/"
-            into "build/docker/"
-            include "*"
-        }
-        copy {
-            from "build/libs"
-            into "build/docker/"
-            include "*.war*"
-        }
+buildscript {
+    repositories {
+        jcenter()
+        mavenLocal()
     }
-    if (OperatingSystem.current().isWindows()) {
-        commandLine 'cmd', '/c', 'docker', 'image', 'build', '-f', 'build/docker/Dockerfile', '-t', '<%= baseName.toLowerCase() %>', 'build/docker/'
-    } else {
-        commandLine 'docker', 'image', 'build', '-f', 'build/docker/Dockerfile', '-t', '<%= baseName.toLowerCase() %>', 'build/docker/'
+
+    dependencies {
+        classpath 'com.bmuschko:gradle-docker-plugin:3.1.0'
     }
 }
+
+apply plugin: com.bmuschko.gradle.docker.DockerRemoteApiPlugin
+
+import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+
+task copyDockerFiles(type: Copy) {
+    description = "Copy dockerfile and required data to build directory"
+    from 'src/main/docker'
+    from "${project.buildDir}/libs"
+    into { "${project.buildDir}/docker" }
+}
+
+task buildDocker(type: DockerBuildImage, dependsOn: 'copyDockerFiles') {
+    description = "Package application as docker image"
+    inputDir = project.file("${project.buildDir}/docker")
+    tags = ["jhipster:${project.version}".toString()]
```

Comment:
a repository can be used here to enable pushing into private/custom container repositories. I would add a commented example for the user, so s/he knows better how to config this before start googling.

Reply:
Good idea. Will add it. Thx.